subprojects/gst-plugins-bad/sys/d3d11/gstd3d11d2d1.cpp

```diff
@@ -287,9 +288,7 @@ gst_d3d11_d2d1_decide_allocation(GstBaseTransform* trans, GstQuery* query)
         return FALSE;
     }
 
-    d3d11_format = gst_d3d11_device_format_from_gst(filter->device,
-        GST_VIDEO_INFO_FORMAT(&vinfo));
-    if (!d3d11_format) {
+    if (!gst_d3d11_device_get_format(filter->device, GST_VIDEO_INFO_FORMAT(&vinfo), d3d11_format)) {
```

Comment:
we actuall pass NULL pointer here ( `GstD3D11Format* d3d11_format = nullptr;` ), where it needs a pointer to allocated memory:
```c
GstD3D11Format d3d11_format;
...
if (!gst_d3d11_device_get_format(filter->device, GST_VIDEO_INFO_FORMAT(&vinfo), &d3d11_format)) {
```

I wonder why it does't crash, because next lines actually try to access the NULL pointer
```c
if (d3d11_format->dxgi_format == DXGI_FORMAT_UNKNOWN) {
```

Reply:
true! just updated. Thanks @aslobodeniuk